### DISCUSSION TITLE
Rule:in has unexpected behavior

### DISCUSSION THREAD

**Discussion by hAz5** [laravel/framework#41618] [Q&A]
in below test, we can see the different answers from a null and empty string
```
 public function testValidateData()
    {
        $rule = new In(['Laravel', 'Framework', 'PHP']);
        $v = new Validator(
            new Translator(new ArrayLoader, 'en'),
            [],
            ['name' => $rule]
        );

        $v->setData(['name' => 'PHP']);
        $this->assertTrue($v->passes());

        $v->setData(['name' => null]);
        $this->assertFalse($v->passes());  //FALSE

        $v->setData(['name' => ""]);//WHY?? TRUE
        $this->assertTrue($v->passes());
        $v->setData(['name' => "   "]);//WHY?? TRUE
        $this->assertTrue($v->passes());
    }

```

**Comment by hAz5** [comment]
@driesvints can you help me with this?

**Reply by driesvints** [reply]
This is because the way the validator works, it considers empty strings as valid values because you didn't explicitly specified that the value is `required`. That's to make sure forms with empty fields can be submitted. Behind the scenes, an `ConvertEmptyStringsToNull` middleware is used to convert these to `null` which the validator doesn't considers as valid.

See https://laravel.com/docs/9.x/validation#a-note-on-optional-fields

**Reply by hAz5** [reply]
thank you @driesvints for your answer, I thought that was a problem because when we migrated from another framework to Laravel we needed to remove that middleware this is why I thought this was a problem.
